pkg/infoprovider/vdpaInfoProvider.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package infoprovider
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+// vdpaInfoProvider is a DeviceInfoProvider that handles the API information of vdpa-capable devices.
+type vdpaInfoProvider struct {
+	dev      types.VdpaDevice
+	vdpaType types.VdpaType
+}
+
+// NewVdpaInfoProvider returns a new InfoProvider associated with the given VDPAInfo
+func NewVdpaInfoProvider(vdpaType types.VdpaType, vdpaDev types.VdpaDevice) types.DeviceInfoProvider {
+	vdpaInfoProvider := &vdpaInfoProvider{
+		dev:      vdpaDev,
+		vdpaType: vdpaType,
+	}
+	return vdpaInfoProvider
+}
+
+// GetDeviceSpecs returns the DeviceSpec slice
+func (vip *vdpaInfoProvider) GetDeviceSpecs() []*pluginapi.DeviceSpec {
+	if healthy, err := vip.isHealthy(); !healthy {
+		glog.Errorf("GetDeviceSpecs(): vDPA is required in the configuration but device does not have a healthy vdpa device: %s",
+			err)
+		return nil
+	}
+	devSpecs := make([]*pluginapi.DeviceSpec, 0)
+
+	// DeviceSpecs only required for vhost vdpa type as the
+	if vip.vdpaType == types.VdpaVhostType {
+		devSpecs = append(devSpecs, &pluginapi.DeviceSpec{
+			HostPath:      vip.dev.GetPath(),
+			ContainerPath: vip.dev.GetPath(),
+			Permissions:   "mrw",
+		})
+	}
+	return devSpecs
+}
+
+// GetEnvVal returns the environment variable value
+func (vip *vdpaInfoProvider) GetEnvVal() string {
```

Comment:
Are `GetEnvVal()` used anywhere? They always return only empty string

Reply:
`GetEnvVal()` called at `(ad *apiDevice) GetMounts()`, before this change it was call in `PciDevice`. `GetMounts()` in turn called at `(rp *ResourcePoolImpl) GetMounts(deviceIDs []string) []*pluginapi.Mount` which is a part of KAPI plugin must implement.